### PR TITLE
Adding Ability To Import Pool

### PR DESCRIPTION
**Pull request by pryorda**
This is to add the ability to import resources and test them. This should not be merged until terraform 0.7 is released.

### CODE REVIEW

File: bigip/resource_bigip_ltm_pool.go [hunk 1]
```diff
@@ -18,13 +18,15 @@ func resourceBigipLtmPool() *schema.Resource {
...
+		Importer: &schema.ResourceImporter{
+			State: resourceBigIpLtmPoolImporter,
+		},
 
 		Schema: map[string]*schema.Schema{
 			"name": &schema.Schema{
 				Type:         schema.TypeString,
 				Required:     true,
 				Description:  "Name of the pool",
-				ForceNew:     true,
```
Comment (aburnett) [inline bigip/resource_bigip_ltm_pool.go:27]: Is changing the name of an existing resource always allowed in via the API? If so we should remove these across the board for consistency. If not or unsure, can you please add a test case to validate changing the name of an existing pool is supported.

**Comment by aburnett** [comment]
How did you update the vendored dependencies? Did you use `govendor update`? Looks like the vendor.json got blown away.

**Comment by aburnett** [comment]
The Terraform test harness also has support for testing import support. Can you please add test cases for the resources which support importing.

Example: https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/import_aws_instance_test.go

**Comment by aburnett** [comment]
Looks like all your importer functions are copy/pastes of `schema.ImportStatePassthrough`. Any reason you don't just use that?

e.g.

```
Importer: &schema.ResourceImporter{
            State: schema.ImportStatePassthrough,
        },
```

File: bigip/resource_bigip_ltm_pool.go (hunk 1 above)
Comment (pryorda) [inline bigip/resource_bigip_ltm_pool.go:27]: We can actually change this to force. I thought i dropped this commit.

**Comment by pryorda** [comment]
On the import passthrough. I made an "assumption" we would add import logic to aide in grabbing additional resources. Say you have and irule attached with a virtualserver or a monitor with a pool. Not sure this is the place for the code as im a new go dev.

**Comment by pryorda** [comment]
I removed the vendor stuff as I didnt think it was needed. However, I do not see a vendor.json in your repo or mine.

File: bigip/resource_bigip_ltm_pool.go (hunk 1 above)
Comment (pryorda) [inline bigip/resource_bigip_ltm_pool.go:27]: Changed this back.

**Comment by aburnett** [comment]
[vendor.json](https://github.com/DealerDotCom/terraform-provider-bigip/blob/master/vendor/vendor.json) 

Vendoring is the closest thing Go has to package management. You shouldn't really be touching anything under the vendor directory by hand. For your PR I would think the only thing you need to do is update the terraform packages. Checkout [govendor](https://github.com/kardianos/govendor) for more info.

Re passthrough: makes sense but not all resources will result in a cascading import (e.g. irule and monitor). Right? For those "leaf" type resources can we just use the built in method?

Let me know if you need any help w/ the vendoring stuff or writing the tests. Thanks!

**Comment by pryorda** [comment]
@aburnett I wont lie, I'm a bit lost on  getting the govendor stuff working. 

As soon as I get this test working, I'll get the leafs updated.

**Comment by pryorda** [comment]
I cant get it to build since I added the vendors back in. Is there something special i need to do?

```
# github.com/pryorda/terraform-provider-bigip/bigip
bigip/resource_bigip_ltm_irule.go:21: unknown schema.Resource field 'Importer' in struct literal
bigip/resource_bigip_ltm_monitor.go:22: unknown schema.Resource field 'Importer' in struct literal
bigip/resource_bigip_ltm_node.go:22: unknown schema.Resource field 'Importer' in struct literal
bigip/resource_bigip_ltm_policy.go:26: unknown schema.Resource field 'Importer' in struct literal
bigip/resource_bigip_ltm_pool.go:23: unknown schema.Resource field 'Importer' in struct literal
bigip/resource_bigip_ltm_virtual_address.go:20: unknown schema.Resource field 'Importer' in struct literal
bigip/resource_bigip_ltm_virtual_server.go:22: unknown schema.Resource field 'Importer' in struct literal
github.com/pryorda/terraform-provider-bigip/vendor/github.com/hashicorp/terraform/state/remote
github.com/pryorda/terraform-provider-bigip/vendor/github.com/hashicorp/terraform/command
github.com/pryorda/terraform-provider-bigip/vendor/github.com/hashicorp/terraform
Makefile:36: recipe for target 'get-deps' failed
make: *** [get-deps] Error 2
dpryor@co007-linux:~/go/src/github.com/pryorda/terraform-provider-bigip$ 
```

**Comment by pryorda** [comment]
@aburnett Have you had a chance to look at this?

**Comment by aburnett** [comment]
Sorry no, I'm going to need a bit.

**Comment by aburnett** [comment]
I'm having some local build issues at the moment, but if you get the vendor directory of your fork back to matching master, what if you then update the terraform deps with

```
find vendor/github.com/hashicorp/terraform -type d -maxdepth 1 | cut -d'/' -f2- | xargs govendor fetch
```

**Comment by pryorda** [comment]
I have it mmatching master now. Ill have to figure out how to update it.

Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Adam Burnett notifications@github.com Date: 6/26/16  8:49 PM  (GMT-07:00) To: DealerDotCom/terraform-provider-bigip terraform-provider-bigip@noreply.github.com Cc: Daniel Pryor daniel@pryorda.net, Author author@noreply.github.com Subject: Re: [DealerDotCom/terraform-provider-bigip] Adding Ability To Import Pool (#26) 
I'm having some local build issues at the moment, but if you get the vendor directory of your fork back to matching master, what if you then update the terraform deps with

find vendor/github.com/hashicorp/terraform -type d -maxdepth 1 | cut -d'/' -f2- | xargs govendor fetch

**Comment by pryorda** [comment]
github.com/masterzen/winrm (download)
package github.com/masterzen/winrm/winrm: cannot find package "github.com/masterzen/winrm/winrm" in any of:
    /usr/lib/go-1.6/src/github.com/masterzen/winrm/winrm (from $GOROOT)
    /home/dpryor/go/src/github.com/masterzen/winrm/winrm (from $GOPATH)
Makefile:36: recipe for target 'get-deps' failed
make: **\* [get-deps] Error 1

Trying to get by this.. :/ Your command worked to update it btw.

**Comment by aburnett** [comment]
Hmm, looks like maybe they refactored that repo (this is why vendoring is good!). Do this:

```
$ go get github.com/masterzen/winrm
$ cd $GOPATH/github.com/masterzen/winrm
$ git checkout 54ea5d01478cfc2afccec1504bd0dfcd8c260cfa
$ cd $GOPATH/<your fork>
$ govendor add github.com/masterzen/winrm/^
```

That's the revision that terraform is using (grabbed from their vendor.json)

**Comment by pryorda** [comment]
You taught me something new. i bought this book to aide my process with learning go :) http://www.gopl.io/. Ill get my stuff updated shortly and get some more import tests going.

**Comment by aburnett** [comment]
@pryorda BTW, you can test all this locally by running `make bin`.

**Comment by pryorda** [comment]
Getting close. 

```
dpryor@co007-linux:~/go/src/github.com/pryorda/terraform-provider-bigip$ make
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestBigipLtmIRule_create
--- SKIP: TestBigipLtmIRule_create (0.00s)
    testing.go:196: Acceptance tests skipped unless env 'TF_ACC' set
=== RUN   TestBigipLtmMonitor_create
--- SKIP: TestBigipLtmMonitor_create (0.00s)
    testing.go:196: Acceptance tests skipped unless env 'TF_ACC' set
=== RUN   TestBigipLtmNode_create
--- SKIP: TestBigipLtmNode_create (0.00s)
    testing.go:196: Acceptance tests skipped unless env 'TF_ACC' set
=== RUN   TestBigipLtmPolicy_create
--- SKIP: TestBigipLtmPolicy_create (0.00s)
    testing.go:196: Acceptance tests skipped unless env 'TF_ACC' set
=== RUN   TestBigipLtmPool_import
--- SKIP: TestBigipLtmPool_import (0.00s)
    testing.go:196: Acceptance tests skipped unless env 'TF_ACC' set
=== RUN   TestBigipLtmPool_create
--- SKIP: TestBigipLtmPool_create (0.00s)
    testing.go:196: Acceptance tests skipped unless env 'TF_ACC' set
=== RUN   TestBigipLtmVA_create
--- SKIP: TestBigipLtmVA_create (0.00s)
    testing.go:196: Acceptance tests skipped unless env 'TF_ACC' set
=== RUN   TestBigipLtmVS_create
--- SKIP: TestBigipLtmVS_create (0.00s)
    testing.go:196: Acceptance tests skipped unless env 'TF_ACC' set
=== RUN   TestValidStringValue
--- PASS: TestValidStringValue (0.00s)
=== RUN   TestInvalidStringValue
--- PASS: TestInvalidStringValue (0.00s)
=== RUN   TestF5NameString
--- PASS: TestF5NameString (0.00s)
=== RUN   TestF5NameSet
--- PASS: TestF5NameSet (0.00s)
=== RUN   TestF5NameList
--- PASS: TestF5NameList (0.00s)
PASS
ok      github.com/pryorda/terraform-provider-bigip/bigip   0.004s
Number of parallel builds: 7

-->       linux/386: github.com/pryorda/terraform-provider-bigip
-->      darwin/386: github.com/pryorda/terraform-provider-bigip
-->    darwin/amd64: github.com/pryorda/terraform-provider-bigip
-->   windows/amd64: github.com/pryorda/terraform-provider-bigip
-->     linux/amd64: github.com/pryorda/terraform-provider-bigip
-->     windows/386: github.com/pryorda/terraform-provider-bigip
dpryor@co007-linux:~/go/src/github.com/pryorda/terraform-provider-bigip$ 
```

Failing one acctest...

```
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestBigipLtmIRule_create
--- PASS: TestBigipLtmIRule_create (4.70s)
=== RUN   TestBigipLtmMonitor_create
--- PASS: TestBigipLtmMonitor_create (7.19s)
=== RUN   TestBigipLtmNode_create
--- PASS: TestBigipLtmNode_create (4.83s)
=== RUN   TestBigipLtmPolicy_create
--- PASS: TestBigipLtmPolicy_create (7.42s)
=== RUN   TestBigipLtmPool_import
--- PASS: TestBigipLtmPool_import (5.35s)
=== RUN   TestBigipLtmPool_create
--- PASS: TestBigipLtmPool_create (5.29s)
=== RUN   TestBigipLtmVA_create
--- PASS: TestBigipLtmVA_create (5.56s)
=== RUN   TestBigipLtmVS_create
{{[]}}
{{[]}}
{{[]}}
{{[]}}
--- FAIL: TestBigipLtmVS_create (6.97s)
    testing.go:255: Step 0 error: Check failed: Check 12/12 error: bigip_ltm_virtual_server.test-vs: Attribute 'policies.3882468147' expected "/Common/test-policy", got ""
=== RUN   TestValidStringValue
--- PASS: TestValidStringValue (0.00s)
=== RUN   TestInvalidStringValue
--- PASS: TestInvalidStringValue (0.00s)
=== RUN   TestF5NameString
--- PASS: TestF5NameString (0.00s)
=== RUN   TestF5NameSet
--- PASS: TestF5NameSet (0.00s)
=== RUN   TestF5NameList
--- PASS: TestF5NameList (0.00s)
FAIL
exit status 1
FAIL    github.com/pryorda/terraform-provider-bigip/bigip   47.321s
Makefile:59: recipe for target 'testacc' failed
make: *** [testacc] Error 1
dpryor@co007-linux:~/go/src/github.com/pryorda/terraform-provider-bigip$ 
```

**Comment by pryorda** [comment]
I have all the imports except the VS but I feel like its not applying the policy to the resource.

https://gist.github.com/pryorda/7d765d3ae78e639754e099d2cd13ad0a

**Comment by pryorda** [comment]
@aburnett Any ideas?

**Comment by aburnett** [comment]
I'm not familiar w/ how the import state tests work, but, looking at [the docs](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestStep) I think `ResourceName` should be the Terraform name of the resource, not the value of the `name` property. So in this case `bigip_ltm_virtual_server.test-vs`. 

What happens if you try that?

**Comment by pryorda** [comment]
It looks like its looking for a non existent policy on the vs.

Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Adam Burnett notifications@github.com Date: 7/7/16  7:07 PM  (GMT-07:00) To: DealerDotCom/terraform-provider-bigip terraform-provider-bigip@noreply.github.com Cc: Daniel Pryor daniel@pryorda.net, Mention mention@noreply.github.com Subject: Re: [DealerDotCom/terraform-provider-bigip] Adding Ability To Import Pool (#26) 
I'm not familiar w/ how the import state tests work, but, looking at the docs I think ResourceName should be the Terraform name of the resource, not the value of the name property. So in this case bigip_ltm_virtual_server.test-vs. 

What happens if you try that?

**Comment by aburnett** [comment]
Agreed. I just noticed the `TestBigipLtmVS_create` is failing also. So something's wrong w/ either creating policies or associating them w/ virtual servers.

**Comment by pryorda** [comment]
Should I set the import test to ignore the policy for now?

**Comment by pryorda** [comment]
@aburnett Status UPdate?

**Comment by aburnett** [comment]
I would recommend we close this PR and create a new one. Please start a fresh branch off latest master and merge in _only_ your changes for import support. The changes to the vendor folder in the current PR make it difficult to get a clear picture of what is actually being submitted (currently shows thousands of files changed). The build & tests need to be happy before we can merge in the PR. I think tidying this up to contain only your work for import support will help in that regard.

Thanks again. This will be awesome to get in!

**Comment by pryorda** [comment]
Closing, recreated with #33

**Pull request closed without merging**